src/main/java/com/innowise/darya/service/SectionServiceImpl.java

```diff
@@ -0,0 +1,50 @@
+package com.innowise.darya.service;
+
+import com.innowise.darya.dto.SectionDTO;
+import com.innowise.darya.entity.Section;
+import com.innowise.darya.exception.ThereIsNoSuchException;
+import com.innowise.darya.repositoty.SectionRepository;
+import com.innowise.darya.transformer.SectionDTOTransformer;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@Transactional
+public class SectionServiceImpl implements SectionService{
+
+    private SectionRepository sectionRepository;
+
+    public SectionServiceImpl(SectionRepository sectionRepository) {
+        this.sectionRepository = sectionRepository;
+    }
+
+    @Override
+    public SectionDTO getSectionStats(Long sectionId) {
+        Optional<Section> section = Optional.ofNullable(sectionRepository.findById(sectionId));
+        if (!section.isPresent()) {
+            throw new ThereIsNoSuchException("section");
+        }
+        SectionDTO sectionDTO = SectionDTOTransformer.SECTION_DTO_TRANSFORMER.sectionToSectionDTO(section.get());
+        return sectionDTO;
+    }
+
+
+
+
+
+    @Override
+    public List<SectionDTO> getAllSection() {
+        List<SectionDTO> sectionDTOList = new ArrayList<>();
+        List<Section> sectionList = sectionRepository.findAll();
+        for (Section section : sectionList) {
```

Comment:
You can create repository method which returns Stream<Entity>. 
After that you can use method map to transform your entity stream to dto stream. after that you can convert this to appropriate collection. 
read about @Transactional(readOnly = true)
https://stackoverflow.com/questions/1614139/spring-transactional-read-only-propagation